Mathlib/Data/Polynomial/FieldDivision.lean

```diff
@@ -655,6 +655,49 @@ theorem isCoprime_of_is_root_of_eval_derivative_ne_zero {K : Type*} [Field K] (f
   rwa [← C_inj, C_0]
 #align polynomial.is_coprime_of_is_root_of_eval_derivative_ne_zero Polynomial.isCoprime_of_is_root_of_eval_derivative_ne_zero
 
+/-- To check a polynomial over a field is irreducible, it suffices to check only for
+divisors that have smaller degree.
+
+See also: `Polynomial.Monic.irreducible_iff_natDegree`.
+-/
+theorem irreducible_iff_degree_lt (p : R[X]) (hp0 : p ≠ 0) (hpu : ¬ IsUnit p) :
+    Irreducible p ↔ ∀ q, q.degree < degree p → q ∣ p → IsUnit q := by
+  rw [← irreducible_mul_C_leadingCoeff_inv hp0,
+      (monic_mul_leadingCoeff_inv hp0).irreducible_iff_degree_lt]
+  simp [hp0]
+  · contrapose! hpu
+    exact isUnit_of_mul_eq_one _ _ hpu
+
+/-- To check a polynomial `p` over a field is irreducible, it suffices to check there are no
+divisors of degree `0 < d ≤ degree p / 2`.
+
+See also: `Polynomial.Monic.irreducible_iff_natDegree'`.
+-/
+theorem irreducible_iff_lt_natDegree_lt {p : R[X]} (hp0 : p ≠ 0) (hpu : ¬ IsUnit p) :
+    Irreducible p ↔ ∀ q, q ∣ p → natDegree q ∉ Finset.Ioc 0 (natDegree p / 2) := by
```

Comment:
This is almost the same as the first theorem on Zulip, except p isn't required to be monic. I don't think it makes much sense to remove `Monic` from the right-hand side; this is meant to be an irreducibility criterion and weakening the RHS only makes it easier to check. If we have that p is irreducible is start with, and q divides p, we know q is a unit or associated to p, so its degree is either 0 or degree p, which we can state as a separate lemma if it looks useful.

Reply:
I'll restate it in terms of your first theorem, adding the monicity condition.